presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -364,7 +364,7 @@ protected Scope visitUse(Use node, Optional<Scope> scope)
         @Override
         protected Scope visitInsert(Insert insert, Optional<Scope> scope)
         {
-            QualifiedObjectName targetTable = createQualifiedObjectName(session, insert, insert.getTarget());
+            QualifiedObjectName targetTable = metadata.redirectTable(session, createQualifiedObjectName(session, insert, insert.getTarget()), warningCollector);
```

Comment:
We might be able to add a helper method in this class
```java
QualifiedObjectName targetTable = getRedirectedTable(insert, insert.getTarget());

private QualifiedObjectName getRedirectedTable(Node node, QualifiedName name) { ... }

Reply:
Hmm this helper method won't help reduce the amount of code so I didn't use it in the updated PR. Please let me know if you think it's still better to use it.